### DISCUSSION TITLE
How can I incorporate trigonometric function into mrgsolve?

### DISCUSSION THREAD

**Discussion by huyo1444** [metrumresearchgroup/mrgsolve#1188] [Q&A]
I want to incorprate circadina rhythem in my PKPD model on mrgsolve.
That's why, I need to express the equation as below:

Y = X1 + X2*cos(2*π*(TIME + X3)/X4)+X5*cos(2*2*π*(TIME + X6)/X7)
(XNs are variables)

I'm appriciate if anyone provied me an answer.

Thanks,

**Comment by kylebaron** [comment]
Hi  @huyo1444  - 

You can write the math like this: 

``` r
library(dplyr)
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union
library(mrgsolve)
#> 
#> Attaching package: 'mrgsolve'
#> The following object is masked from 'package:stats':
#> 
#>     filter

code <- '

$PARAM X1 = 50, X2 = 4, X3 = -2, X4 = 12, X5 = 2, X6 = 4, X7 = 19

$ERROR
double A = X2 * cos(2 * M_PI * (TIME + X3)/X4);
double B = X5 * cos(2 * M_PI * (TIME + X6)/X7);

capture Y = X1 + A + B;

'

mod <- mcode("circ", code)
#> Building circ ...
#> done.

mrgsim(mod, end = 72, delta = 0.1) %>% plot()
```

![](https://i.imgur.com/UHu8SlN.png)

<sup>Created on 2024-04-29 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>


If there are other components that need to get hooked up in the model, you'll have to give us more information on that. 

Kyle

**Reply by huyo1444** [reply]
Hi @kylebaron 

Thank you very much for your reply!
I could do my ideal simulation.

Then, I have an additional question.
I want to simulate drug effect with circadian rhythm.
So I want to set specific time (e.x., AM10:00) as dosing time. How can I do that with mrgsolve?

I'm looking forward to your kind reply.

Thanks,
